Update remote-debugging docs to include hololens instructions (#2547)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/how-to/remote-debugging.md

Now you can use Microsoft Edge DevTools to inspect and debug the WebView2 control.

<!-- ====================================================================== -->
## Debugging HoloLens 2 WebView2 WinUI 2 (UWP) apps

WebView2 on HoloLens 2 and the WebView2 plugin for Unity are both in Preview and are subject to change before general availability.  The WebView2 Preview is available in the Insider Preview for Microsoft HoloLens.  

> To access this preview, you must be enrolled in the Windows Insider Program; see [Start receiving Insider builds](https://learn.microsoft.com/en-us/hololens/hololens-insider#start-receiving-insider-builds) in _Insider preview for Microsoft HoloLens_.

Reviewer: ```suggestion
To access this preview, you must be enrolled in the Windows Insider Program; see [Start receiving Insider builds](/hololens/hololens-insider#start-receiving-insider-builds) in _Insider preview for Microsoft HoloLens_.
```

Now you can use Microsoft Edge DevTools to inspect and debug the WebView2 control.

<!-- ====================================================================== -->
## Debugging Hololens 2 WebView2 WinUI 2 (UWP) apps

Attach Microsoft Edge DevTools remotely to a Hololens 2 WebView2 WinUI 2 (UWP) app as follows:

> <!-- steps 1 & 2: -->

Reviewer: Global: If the correct steps got copied from main section to here, probably now remove hidden comments like:
```
<!-- steps 1 & 2: -->
<!-- / steps 1 & 2 -->
<!-- steps 8-12: -->
```
which are probably 1 too high now anyway

Now you can use Microsoft Edge DevTools to inspect and debug the WebView2 control.

<!-- ====================================================================== -->
## Debugging Hololens 2 WebView2 WinUI 2 (UWP) apps

Attach Microsoft Edge DevTools remotely to a Hololens 2 WebView2 WinUI 2 (UWP) app as follows:

1. Follow steps 1 & 2 under [Debugging desktop WebView2 WinUI 2 (UWP) apps](#debugging-desktop-webview2-winui-2-uwp-apps).

> 1. Enable Device Portal: Open Windows **Settings**, select **Update & Security** > **For developers**, and then turn on the **Developer Portal** toggle.

Reviewer: within any list, either have 1 or 2 spaces after period.  In a list, the indent amount of a png must match that of the text.  I can check it out in VS Code locally and post changes if needed.  Or do a PR into this PR.

> 1. Enable Device Portal: Open Windows **Settings**, select **Update & Security** > **For developers**, and then turn on the **Developer Portal** toggle.

Reviewer: ```suggestion
1. Enable Device Portal: Open Windows **Settings**, select **Update & Security** > **For developers**, and then turn on the **Device Portal** toggle:
```

>     ![Enable Device Portal](./remote-debugging-images/enable-device-portal-holo.png)

Reviewer: ```suggestion
   ![Enable Device Portal](./remote-debugging-images/enable-device-portal-holo.png)
```
or, add two spaces after each "1." in this list

>     After Device Portal has been enabled, note the URL that will be used for the remote debugging connection. The URL will be under **Connect using** and **Ethernet**.

Reviewer: ```suggestion
   After Device Portal has been enabled, note the URL that will be used for the remote debugging connection. The URL will be under **Connect using** and **Ethernet**.
```

1. Install [Remote Tools for Microsoft Edge](https://www.microsoft.com/store/productId/9P6CMFV44ZLT) from the Microsoft Store on your Hololens 2 device.

1. Follow steps 8-12 under [Debugging desktop WebView2 WinUI 2 (UWP) apps](#debugging-desktop-webview2-winui-2-uwp-apps).

1. Launch your app on your Hololens 2 device.

> 1. Follow steps 14 & 15 under [Debugging desktop WebView2 WinUI 2 (UWP) apps](#debugging-desktop-webview2-winui-2-uwp-apps).

Reviewer: global: try to avoid numbered steps, the numbering is subject to change.  I plan to edit the .md locally to find a solution.

1. Verify that you can successfully connect, so that your debuggable WebView2 control, named **Edge**, is listed under your machine name:

>     ![Connect to WebView2 on Hololens 2](./remote-debugging-images/connect-to-webview2-hololens.png)

Reviewer: ```suggestion
   ![Connect to WebView2 on Hololens 2](./remote-debugging-images/connect-to-webview2-hololens.png)
```

> 1. Follow steps 1 & 2 under [Debugging desktop WebView2 WinUI 2 (UWP) apps](#debugging-desktop-webview2-winui-2-uwp-apps).
> 
> 1. Enable Device Portal: Open Windows **Settings**, select **Update & Security** > **For developers**, and then turn on the **Developer Portal** toggle.
> 
>     ![Enable Device Portal](./remote-debugging-images/enable-device-portal-holo.png)
> 
>     After Device Portal has been enabled, note the URL that will be used for the remote debugging connection. The URL will be under **Connect using** and **Ethernet**.
> 
> 1. Install [Remote Tools for Microsoft Edge](https://www.microsoft.com/store/productId/9P6CMFV44ZLT) from the Microsoft Store on your Hololens 2 device.
> 
> 1. Follow steps 8-12 under [Debugging desktop WebView2 WinUI 2 (UWP) apps](#debugging-desktop-webview2-winui-2-uwp-apps).
> 
> 1. Launch your app on your Hololens 2 device.
> 
> 1. Follow steps 14 & 15 under [Debugging desktop WebView2 WinUI 2 (UWP) apps](#debugging-desktop-webview2-winui-2-uwp-apps).
> 
> 1. Verify that you can successfully connect, so that your debuggable WebView2 control, named **Edge**, is listed under your machine name:
> 
>     ![Connect to WebView2 on Hololens 2](./remote-debugging-images/connect-to-webview2-hololens.png)
> 
> 1. Follow step 17 under [Debugging desktop WebView2 WinUI 2 (UWP) apps](#debugging-desktop-webview2-winui-2-uwp-apps).
> 
> Now you can use Microsoft Edge DevTools to inspect and debug the WebView2 control.

Reviewer: ```suggestion
1.  Install [Microsoft Edge Canary Channel](https://www.microsoftedgeinsider.com/download/canary).

1.  In Microsoft Edge Canary, go to to `edge://inspect` and verify that the **Inspect with Edge Developer Tools** page exists in your version of the browser:

    ![DevTools Inspect utility page](./remote-debugging-images/inspect-devtools-page-supported.png)

1.  Enable Device Portal: Open Windows **Settings**, select **Update & Security** > **For developers**, and then turn on the **Device Portal** toggle:

    ![Enable Device Portal](./remote-debugging-images/enable-device-portal-holo.png)

    In **Connect using** > **Ethernet**, note the URL that will be used for the remote debugging connection.

1.  Install [Remote Tools for Microsoft Edge](https://www.microsoft.com/store/productId/9P6CMFV44ZLT) from the Microsoft Store on your Hololens 2 device.

1.  In Microsoft Edge Canary, go to the Device Portal URL that you previously noted.  Verify that the **Confirm security settings** page is loaded (in the **Windows Device Portal** tab):

    ![The 'Confirm security settings' page in the 'Windows Device Portal' tab](./remote-debugging-images/open-device-portal.png)

1.  If you are only debugging apps that are running locally, you can select the **Continue with an unsecured connection** button.

    Or, if you aren't running the app locally, or if you need to connect over HTTPS, select the **Configure a secure connection** link, and then follow the steps in the page that you're redirected to.

1.  Verify that you're redirected to the **Apps manager: Windows Device Portal** page:

    ![Device Portal](./remote-debugging-images/device-portal.png)

1.  Go to `http://<Device Portal URL>/msedge` and verify Remote Tools for Microsoft Edge are functioning. The page loads an empty list: `[]`, unless there's an application running that is debuggable by remote tools.

1.  Configure your WebView2 WinUI 2 (UWP) app for remote debugging:

    To enable remote debugging, two environment variables must be set in your project.  These variables must be set before creating the `CoreWebView2` instance, and before either setting the `WebView2.Source` property or calling the `WebView2.EnsureCoreWebView2Async` method.

    Set the following variables:

    ```
    "WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS", "--enable-features=msEdgeDevToolsWdpRemoteDebugging"
    "WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS", "WEBVIEW2_RELEASE_CHANNEL_PREFERENCE=1"
    ```
    
    For example, if you're using the [WebView2 Sample UWP app](https://github.com/MicrosoftEdge/WebView2Samples/tree/main/SampleApps/webview2_sample_uwp), you can set the environment variables by adding the following lines in the `Browser.xaml.cs` file:
    
    ```csharp
    Environment.SetEnvironmentVariable("WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS", "--enable-features=msEdgeDevToolsWdpRemoteDebugging");
    Environment.SetEnvironmentVariable("WEBVIEW2_ADDITIONAL_BROWSER_ARGUMENTS", "WEBVIEW2_RELEASE_CHANNEL_PREFERENCE=1");
    ```

    The lines are numbered 33 and 34 in this screenshot, in the `Browser()` constructor, below an `#endif` that wraps an existing `SetEnvironmentVariable` statement:

    ![Browser Arguments Remote Debugging](./remote-debugging-images/browser-arguments-remote-debugging.png)

1.  Launch your app on your Hololens 2 device.

1.  Go to `http://<Device Portal URL>/msedge`, such as `http://localhost:50080/msedge`, and make sure your WebView2 instance appears in the list:

    ![Debuggable WebView2 Instance](./remote-debugging-images/debuggable-webviews.png)

1.  Go to `edge://inspect`.  In the **Connect to a remote Windows device** text box, enter `http://<Device Portal URL>`, such as `http://localhost:50080`, and then click **Connect to Device**.

1.  Verify that you can successfully connect, so that your debuggable WebView2 control, named **Edge**, is listed under your machine name:

    ![Connect to WebView2 on Hololens 2](./remote-debugging-images/connect-to-webview2-hololens.png)

1.  At the bottom of the **Edge** WebView2 control entry, click the **inspect** link.  Microsoft Edge DevTools opens for the WebView2 control:

    ![DevTools Inspect Window](./remote-debugging-images/devtools-inspect.png)

Now you can use Microsoft Edge DevTools to inspect and debug the WebView2 control in your Hololens 2 WebView2 WinUI 2 (UWP) app.
```
